### PR TITLE
add optional avro protocol

### PR DESCRIPTION
**Pull request by jfromaniello**
The AVRO implementation has proven to be faster than protocol-buffers.

You can compare by running
```
#protocol-buffers:
npm run perf

#protocol-buffers:
npm run perf -- --avro
```

This pull request add an optional support for the avro protocol, the default is still protocol buffers.

### CODE REVIEW

File: client.js [hunk 1]
```diff
@@ -82,19 +114,53 @@ LimitdClient.prototype._request = function (request, type, done) {
...
+  this.stream.write(request);
 
   if (!done) return;
 
   this.once('response_' + request.id, function (response) {
-    if (response['.limitd.ErrorResponse.response'] &&
-        response['.limitd.ErrorResponse.response'].type === ErrorResponse.Type.UNKNOWN_BUCKET_TYPE) {
-      return done(new Error(type + ' is not a valid bucket type'));
+    if (client._options.protocol === 'protocol-buffers') {
+      if (response['.limitd.ErrorResponse.response'] &&
```
Comment (dschenkelman) [inline client.js]: looks weird. how about strategies for each protocol to avoid all ifs?

File: client.js [hunk 2]
```diff
@@ -135,40 +196,39 @@ LimitdClient.prototype.put = function (type, key, count, done) {
...
 
 LimitdClient.prototype.status = function (type, key, done) {
-  var request = new RequestMessage({
+  var request = {
     'id':     randomstring.generate(7),
     'type':   type,
     'key':    key,
-    'method': RequestMessage.Method.STATUS,
-  });
+    'method': 'STATUS',
```
Comment (dschenkelman) [inline client.js]: constants module

File: lib/handlers/status.js [hunk 3]
```diff
@@ -1,7 +1,46 @@
-var Response = require('../../messages').Response;
-var StatusResponse = require('../../messages').StatusResponse;
+var Response = require('../../messages/protocol_buffers').Response;
+var StatusResponse = require('../../messages/protocol_buffers').StatusResponse;
 
-module.exports.handle = function (buckets, log, message, done) {
+function build_status_response (protocol, message, items) {
+  if (protocol === 'protocol-buffers') {
```
Comment (dschenkelman) [inline lib/handlers/status.js]: this should be part of the strategies

**Comment by jfromaniello** [comment]
Thanks for the feedback I will solve these issues in https://github.com/limitd/node-client

**Pull request merged**